### ISSUE TITLE
No model .t7 file being created while training on own data

### ISSUE THREAD

**Issue by eldor4do** [karpathy/neuraltalk2#71]
I'm running the train.lua file as,
`$ th train.lua -input_h5 test_h5.h5 -input_json test_json.json -gpuid -1 -max_iters 5`

I'm getting output as,
`DataLoader loading json file: 	test_json.json	
vocab size is 13	
DataLoader loading h5 file: 	test_h5.h5	
read 101 images of size 3x256x256	
max sequence length in data is 16	
assigned 71 images to split train	
assigned 20 images to split val	
assigned 10 images to split test	
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:505] Reading dangerously large protocol message.  If the message turns out to be larger than 1073741824 bytes, parsing will be halted for security reasons.  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:78] The total number of bytes read was 553432081
Successfully loaded model/VGG_ILSVRC_16_layers.caffemodel`

Some more output comes, after this, the end output is like,
`evaluating validation performance... 12/20 (2.626283)	
validation loss: 	2.6253916986287	
nil	
wrote json checkpoint to model_id.json	
iter 1: 2.450893	
iter 2: 2.310778	
iter 3: 2.097281	
iter 4: 1.869632`

That is it. There is no ''wrote checkpoint to..' being printed, neither any .t7 file is created. I only get the model_id.json

**Comment by liuchang8am** [comment]
check the -save_checkpoint_every parameter.

**Comment by LuoweiZhou** [comment]
Hi, have you solved it? I got a similar problem...

**Comment by eldor4do** [comment]
@LuoweiZhou If you're not using the -save_checkoint_every parameter, then make a change in this,
https://github.com/karpathy/neuraltalk2/blob/master/train.lua#L302
There is a problem in that code, make `iter == opt.max_iters` to `iter == opt.max_iters-1`

**Comment by polyrhythmatic** [comment]
@eldor4do I would like to encourage you to make a PR for this. I would if I felt confident enough to do this. I just lost a nights worth of training assuming that there was a default value because of this!
